pkg/filters/proxy/server.go

```diff
@@ -31,6 +34,14 @@ type Server struct {
 	Weight         int      `json:"weight" jsonschema:"omitempty,minimum=0,maximum=100"`
 	KeepHost       bool     `json:"keepHost" jsonschema:"omitempty,default=false"`
 	addrIsHostName bool
+	health         atomic.Value
```

Comment:
is it necessary to use `atomic.Value`? it seems there are only one goroutine that accesses this field.